rfcs/0088-commit-message-guideline.md

```diff
@@ -0,0 +1,187 @@
+“Fix flaky test” and without any body text should be avoided. Also, for the
+review, it will leave the reviewer wondering about what exactly was fixed or
+changed and why the change is necessary, slowing the review.
+
+Below is an example that can be used as a model:
+
+> [microTVM] Zephyr: Remove zephyr_board option from build, flash, and open_transport methods
+>
+> Currently it’s necessary to pass the board type via ‘zephyr_board’ option to
+> the Project API build, flash, and open_transport methods.
+>
+> However, since the board type is already configured when the project is
+> created (i.e. when the generate_project method is called), it’s possible to
+> avoid this redundancy by obtaining the board type from the project
+> configuration files.
+>
+> This commit adds code to obtain the board type from the project CMake files,
+> removing this option from build, flash, and open_transport methods, so it’s
+> only necessary to specify the ‘zephyr_board’ option when calling
+> generate_project.
+>
+> This commit also moves the ‘verbose’ and ‘west_cmd’ options from ‘build’
+> method to ‘generate_project’, reducing further the number of required options
+> when building a project, since the ‘build’ method is usually called more often
+> than the ‘generate_project’.
+
+After a new PR is created and the review starts it’s common that reviewers will
+request changes. Usually the author will address the reviewers’ comments and
+push additional commits on top of the initial ones. For these additional commits
+there is no recommendation regarding the commit messages. However if the
+additional commits render the PR title and/or body outdated then it's the
+author's responsibility to keep the PR title and body in sync with new changes
+in the code and updated the PR title and body accordingly (remember that the PR
+title and body will be used to compose the final commit message that will land
+in the main tree).
+
+Committers will seek to fix any issues with the commit message prior to
+committing but they retain the right to inform the author of the rules and
+encourage them to follow them in future. Also, they retain the right to ask to
+the author to update the PR title and/or body when they are not correctly
+updated or fixed.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+TVM Community must reach a certain concensus about the rules in this guideline,
+hence this RFC will be voted.
+
+Once it's voted and approved the Commit Message Guideline text will be added to
+`./docs/contribute/pull_request.rst` doc, under section 'Submit a Pull Request',
+below subsection 'Guidelines', as a subsection named “Commit Message Guideline”.
+The text in the second-last item in subsection 'Guidelines' that mentions PR
+tags will also be extended (a hyperlink will be added) to refer to this
+guideline, since it also contains guidelines about use of tags.
```

Comment:
My takes: I would love to recommend that those rules be numbered, and written in one concise sentence with an positive/negative example each, so those could be the items the reviewers refer to.

For example:

```Rule R1. PR Title
- R1.1. Use uppercase for the first letter of the PR title, as well as acronyms, but anything else should be lowercased.
✅ This is a PR title
❌ THIS IS a pR tiTLE
- R1.2. Do not use any period at the end of the title.
✅ PR title without period
❌ PR title with a period.
```

I'm happy to work together on drafting this if you guys think it's a right direction to go :-)

Reply:
I think it would be great too, as the list as you describe is easier to point out in reviews, and also for authors to checklist their patches.

I expect "the list" to contain just high-priority set of enforced items and few recommendations, so that we can focus what's essential. I don't expect it to be a long document, if we can make it as a paragraph and a couple of small lists. I'd like to make it as easier and accessible as possible for new contributors to understand quickly.

Reply:
Marking this a resolved. Please reopen in case there are any additional point.

Reply:
> I expect "the list" to contain just high-priority set of enforced items and few recommendations, so that we can focus what's essential. I don't expect it to be a long document, if we can make it as a paragraph and a couple of small lists. I'd like to make it as easier and accessible as possible for new contributors to understand quickly.

100% agreed - I hated reading long documents too. An agile OSS should prefer concise guidelines that are to-the-point, instead of tedious unnecessarily long ones :-)